src/controllers/cgnController.ts

```diff
@@ -12,19 +12,34 @@ import {
   IResponseErrorValidation,
   IResponseSuccessAccepted,
   IResponseSuccessJson,
-  IResponseSuccessRedirectToResource
+  IResponseSuccessRedirectToResource,
+  ResponseErrorForbiddenNotAuthorized
 } from "italia-ts-commons/lib/responses";
 
 import { EycaActivationDetail } from "generated/io-cgn-api/EycaActivationDetail";
 import { EycaCard } from "generated/io-cgn-api/EycaCard";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
 import { Card } from "../../generated/cgn/Card";
 import CgnService from "../../src/services/cgnService";
 import { InstanceId } from "../../generated/cgn/InstanceId";
-import { withUserFromRequest } from "../types/user";
+import { User, withUserFromRequest } from "../types/user";
 import { CgnActivationDetail } from "../../generated/io-cgn-api/CgnActivationDetail";
 
+export const withAllowedUser = async <T>(
+  user: User,
+  allowedFiscalCodes: ReadonlyArray<FiscalCode>,
+  f: (user: User) => Promise<T>
+) =>
+  allowedFiscalCodes.includes(user.fiscal_code) ||
+  allowedFiscalCodes.length === 0
```

Comment:
Invert conditions, it's more efficient
```suggestion
  allowedFiscalCodes.length === 0 ||
  allowedFiscalCodes.includes(user.fiscal_code)
```